packages/data-point/src/DataPoint.js

```diff
@@ -0,0 +1,125 @@
+const { createReducer } = require("./create-reducer");
+const { Accumulator } = require("./Accumulator");
+const { resolve } = require("./resolve");
+const { Cache } = require("./Cache");
+const isPlainObject = require("./is-plain-object");
+
+/**
+ * Applies a reducer from an accumulator object.
+ *
+ * @param {Accumulator} acc accumulator object to be used.
+ * @param {Reducer} reducer
+ * @returns {Promise<any>} resolved value
+ */
+async function resolveFromAccumulator(acc, reducer) {
+  const parsedReducers = createReducer(reducer);
+  return resolve(acc, parsedReducers);
+}
+
+/**
+ * Applies a reducer to the provided input value.
+ *
+ * @param {any} input value to process.
+ * @param {Reducer} reducer valid reducer to process the value.
+ * @param {Object} options
+ * @param {Cache|undefined} options.cache cache manager, see Cache for options.
+ * @param {Object|undefined} options.locals persistent object that is
+ * accessible via the Accumulator object on every reducer.
+ * @param {OpenTrace.Span|undefined} options.tracer when provided it should
+ * comply with the **opentracing** Span API.
+ * @returns {Promise<any>} resolved value
+ */
+async function resolveFromInput(input, reducer, options = {}) {
+  const acc = new Accumulator({
+    value: input,
+    locals: options.locals,
+    resolve: resolveFromAccumulator,
+    cache: options.cache,
+    tracer: options.tracer
+  });
+
+  return resolveFromAccumulator(acc, reducer);
+}
+
+/**
+ * @param {Object} locals
+ * @throws Error if the value is not a simple object. To kow which values are
+ * considered as simple objects please read the unit test for `isPlainObject`.
```

Comment:
it wouldn't hurt to mention "simple" or "plain" in the error message

`"'options.locals' must be undefined or a plain object"`